keyboards/ut472/keymaps/hvp/keymap.c

```diff
@@ -0,0 +1,207 @@
+  _______, _______,  _______,  RGB_HUD,  RGB_HUI,  _______,  _______,  _______,  _______,  _______,  _______,  _______,
+  KC_MS_L, KC_MS_D, KC_MS_U, KC_MS_R, _______,     _______,      _______, KC_HOME, KC_PGDN, KC_PGUP, KC_END
+),
+
+  /* FN Layer 2
+   * ,-------------------------------------------------------------------------.
+   * |   ~  |  !  |  @  |  #  |  $  |  %  |  ^  |  &  |  *  |  (  |  )  |Delete|
+   * |-------------------------------------------------------------------------+ 
+   * |      |     |     |     |     |     |     |  _  |  +  |  { |  }  |   |   |
+   * |-------------------------------------------------------------------------+
+   * |       | F1  | F2  | F3  | F4  | F5  | F6  | F7  | F8  | F9  | F10 |     |
+   * |-------------------------------------------------------------------------+
+   * |     |     |     |Capsl|      |          |       | Home| PgDn| PgUp| End |
+   * `-------------------------------------------------------------------------'
+   */
+
+LAYOUT( /* Left */
+  KC_TILDE,  KC_EXCLAIM,  KC_AT,  KC_HASH,  KC_DOLLAR, KC_PERCENT, KC_CIRCUMFLEX, KC_AMPERSAND, KC_ASTERISK, KC_LEFT_PAREN, KC_RIGHT_PAREN, KC_BSPC,
+  KC_DELETE, _______, _______, _______, _______, _______, _______, KC_UNDERSCORE, KC_PLUS, KC_LEFT_CURLY_BRACE, KC_RIGHT_CURLY_BRACE, KC_PIPE,
+  _______, _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,  _______,
+  _______, _______, _______, _______, _______,     _______,      _______, KC_HOME, KC_PGDN, KC_PGUP, KC_END
+),
+
+  /* FN Layer 2
+   * ,-------------------------------------------------------------------------.
+   * | Esc | Calc|Webhm| Mail| Comp|     |     |     |     |     |PrtSc|       |
+   * |-------------------------------------------------------------------------+
+   * |      |     |     |     |     |     |     |     |     |     |     |      |
+   * |-------------------------------------------------------------------------+
+   * |       |    |     |     |     |     |     |     |     |     |     |     |
+   * |-------------------------------------------------------------------------+
+   * |     |     |     |     |      |          |       |MousL|MousD|MousU|MousR|
+   * `-------------------------------------------------------------------------'
+   */
+
+LAYOUT( /* Tab */
+  KC_F1,  KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12,
+  _______, _______, _______, _______, _______, _______, KC_LEFT, KC_DOWN, KC_UP, KC_RGHT, _______, KC_PSCR,
+  _______, _______, _______, _______, _______, _______, KC_HOME, KC_PGDN, KC_PGUP, KC_END, _______, _______,
+  RESET, _______, _______, _______, KC_MUTE,     KC_MPLY,      KC_MSTP, KC_MPRV, KC_VOLD, KC_VOLU, KC_MNXT
+),
+};
+
+// Tap dance 1
+void dance_1_finished(qk_tap_dance_state_t *state, void *user_data)
+{
+  if (state->count == 2)
+  {
+    register_code(KC_SCLN);
```

Comment:
A lot of these could be replaced with `tap_code`, `register_code16(RALT(kc))`, etc.